Импорты

In [144]:
import pandas as pd
import numpy as np

Константные переменные

In [145]:
INDEX_COL = 0
AVERAGE_BILL = "average_bill"
AVERAGE_BILL_FILLNA =  "average_bill_filna"
RUBRIC = "rubric"
COFFEE_TO_GO = "coffee_to_go"

In [146]:
data = pd.read_csv("datasets_eat_places.csv", index_col=INDEX_COL)

In [147]:
data

org_id city  average_bill    rating           rubric  \
1      16076540698036998306  msk         500.0  4.375000  Быстрое питание   
2       8129364761615040323  msk         500.0  4.000000          Кофейня   
4      13418544315327784420  msk         500.0  4.409091          Кофейня   
5       4997872545011433005  msk         500.0  4.541667          Кофейня   
8      11913692756639492916  spb         500.0  3.600000         Столовая   
...                     ...  ...           ...       ...              ...   
68333  17936653153885963296  msk           NaN  4.487952         Ресторан   
68334   4379286080707082909  msk           NaN  3.812500             Кафе   
68335   7916477189329738565  msk           NaN  4.894231         Ресторан   
68336  12358902585434046825  msk           NaN  4.156250             Кафе   
68338    125079240973229511  spb           NaN  4.633333         Ресторан   

       coffee_to_go  food_delivery  breakfast  summer_terrace  wi_fi  \
1                 1              1          0               0      0   
2                 1              1          1               1      1   
4                 1              1          1               1      1   
5                 1              1          1               1      1   
8                 0              1          0               1      0   
...             ...            ...        ...             ...    ...   
68333             1              1          1               1      1   
68334             0              1          0               1      1   
68335             1              1          1               1      0   
68336             1              1          1               1      0   
68338             0              0          0               0      1   

       business_lunch  payment_by_credit_card  car_park  gift_certificate  \
1                   0                       1         0                 0   
2                   1                       0         0                 0   
4                   1                       0         0                 0   
5                   1                       1         1                 0   
8                   0                       1         0                 1   
...               ...                     ...       ...               ...   
68333               1                       1         0                 0   
68334               1                       1         0                 1   
68335               1                       1         0                 1   
68336               1                       1         0                 1   
68338               0                       1         1                 0   

       karaoke  free_delivery  music  pets  air_conditioning  \
1            0              0      0     0                 0   
2            0              0      0     0                 0   
4            0              0      0     0                 0   
5            0              0      0     0                 0   
8            0              0      0     0                 0   
...        ...            ...    ...   ...               ...   
68333        1              0      0     0                 0   
68334        1              0      0     0                 0   
68335        1              1      0     0                 0   
68336        1              1      0     0                 0   
68338        0              0      0     0                 0   

       view_on_ostankino_tower  
1                            0  
2                            0  
4                            0  
5                            1  
8                            0  
...                        ...  
68333                        0  
68334                        0  
68335                        0  
68336                        0  
68338                        0  

[43195 rows x 20 columns]

Задание номер 2


In [148]:
data.head()



org_id city  average_bill    rating           rubric  \
1  16076540698036998306  msk         500.0  4.375000  Быстрое питание   
2   8129364761615040323  msk         500.0  4.000000          Кофейня   
4  13418544315327784420  msk         500.0  4.409091          Кофейня   
5   4997872545011433005  msk         500.0  4.541667          Кофейня   
8  11913692756639492916  spb         500.0  3.600000         Столовая   

   coffee_to_go  food_delivery  breakfast  summer_terrace  wi_fi  \
1             1              1          0               0      0   
2             1              1          1               1      1   
4             1              1          1               1      1   
5             1              1          1               1      1   
8             0              1          0               1      0   

   business_lunch  payment_by_credit_card  car_park  gift_certificate  \
1               0                       1         0                 0   
2               1                       0         0                 0   
4               1                       0         0                 0   
5               1                       1         1                 0   
8               0                       1         0                 1   

   karaoke  free_delivery  music  pets  air_conditioning  \
1        0              0      0     0                 0   
2        0              0      0     0                 0   
4        0              0      0     0                 0   
5        0              0      0     0                 0   
8        0              0      0     0                 0   

   view_on_ostankino_tower  
1                        0  
2                        0  
4                        0  
5                        1  
8                        0

In [149]:
counts = 0
for i, j in data.iterrows():
    if j[AVERAGE_BILL] > 2500 or j[AVERAGE_BILL] == np.nan:
        counts += 1
print(counts)

483


In [150]:
data[AVERAGE_BILL]

1        500.0
2        500.0
4        500.0
5        500.0
8        500.0
         ...  
68333      NaN
68334      NaN
68335      NaN
68336      NaN
68338      NaN
Name: average_bill, Length: 43195, dtype: float64

In [151]:
data[AVERAGE_BILL_FILLNA] = data[AVERAGE_BILL]
data[AVERAGE_BILL_FILLNA].fillna(0, inplace=True)
avarage_price = sum(data[AVERAGE_BILL_FILLNA] / len(data[AVERAGE_BILL_FILLNA])) #средняя цена

In [152]:
avarage_price

564.3593008449949

In [153]:
data[AVERAGE_BILL].fillna(avarage_price, inplace=True)

In [154]:
data

org_id city  average_bill    rating           rubric  \
1      16076540698036998306  msk    500.000000  4.375000  Быстрое питание   
2       8129364761615040323  msk    500.000000  4.000000          Кофейня   
4      13418544315327784420  msk    500.000000  4.409091          Кофейня   
5       4997872545011433005  msk    500.000000  4.541667          Кофейня   
8      11913692756639492916  spb    500.000000  3.600000         Столовая   
...                     ...  ...           ...       ...              ...   
68333  17936653153885963296  msk    564.359301  4.487952         Ресторан   
68334   4379286080707082909  msk    564.359301  3.812500             Кафе   
68335   7916477189329738565  msk    564.359301  4.894231         Ресторан   
68336  12358902585434046825  msk    564.359301  4.156250             Кафе   
68338    125079240973229511  spb    564.359301  4.633333         Ресторан   

       coffee_to_go  food_delivery  breakfast  summer_terrace  wi_fi  ...  \
1                 1              1          0               0      0  ...   
2                 1              1          1               1      1  ...   
4                 1              1          1               1      1  ...   
5                 1              1          1               1      1  ...   
8                 0              1          0               1      0  ...   
...             ...            ...        ...             ...    ...  ...   
68333             1              1          1               1      1  ...   
68334             0              1          0               1      1  ...   
68335             1              1          1               1      0  ...   
68336             1              1          1               1      0  ...   
68338             0              0          0               0      1  ...   

       payment_by_credit_card  car_park  gift_certificate  karaoke  \
1                           1         0                 0        0   
2                           0         0                 0        0   
4                           0         0                 0        0   
5                           1         1                 0        0   
8                           1         0                 1        0   
...                       ...       ...               ...      ...   
68333                       1         0                 0        1   
68334                       1         0                 1        1   
68335                       1         0                 1        1   
68336                       1         0                 1        1   
68338                       1         1                 0        0   

       free_delivery  music  pets  air_conditioning  view_on_ostankino_tower  \
1                  0      0     0                 0                        0   
2                  0      0     0                 0                        0   
4                  0      0     0                 0                        0   
5                  0      0     0                 0                        1   
8                  0      0     0                 0                        0   
...              ...    ...   ...               ...                      ...   
68333              0      0     0                 0                        0   
68334              0      0     0                 0                        0   
68335              1      0     0                 0                        0   
68336              1      0     0                 0                        0   
68338              0      0     0                 0                        0   

       average_bill_filna  
1                   500.0  
2                   500.0  
4                   500.0  
5                   500.0  
8                   500.0  
...                   ...  
68333                 0.0  
68334                 0.0  
68335                 0.0  
68336                 0.0  
68338                 0.0  

[43195 rows x 21 columns]

Легко заметить посмотрев на предыдущие таблицы, что NaN мы убрали из среднего чека

In [155]:
data[AVERAGE_BILL] = data[AVERAGE_BILL].map(lambda x: avarage_price if (x > 2500) else x)

In [156]:
data

org_id city  average_bill    rating           rubric  \
1      16076540698036998306  msk    500.000000  4.375000  Быстрое питание   
2       8129364761615040323  msk    500.000000  4.000000          Кофейня   
4      13418544315327784420  msk    500.000000  4.409091          Кофейня   
5       4997872545011433005  msk    500.000000  4.541667          Кофейня   
8      11913692756639492916  spb    500.000000  3.600000         Столовая   
...                     ...  ...           ...       ...              ...   
68333  17936653153885963296  msk    564.359301  4.487952         Ресторан   
68334   4379286080707082909  msk    564.359301  3.812500             Кафе   
68335   7916477189329738565  msk    564.359301  4.894231         Ресторан   
68336  12358902585434046825  msk    564.359301  4.156250             Кафе   
68338    125079240973229511  spb    564.359301  4.633333         Ресторан   

       coffee_to_go  food_delivery  breakfast  summer_terrace  wi_fi  ...  \
1                 1              1          0               0      0  ...   
2                 1              1          1               1      1  ...   
4                 1              1          1               1      1  ...   
5                 1              1          1               1      1  ...   
8                 0              1          0               1      0  ...   
...             ...            ...        ...             ...    ...  ...   
68333             1              1          1               1      1  ...   
68334             0              1          0               1      1  ...   
68335             1              1          1               1      0  ...   
68336             1              1          1               1      0  ...   
68338             0              0          0               0      1  ...   

       payment_by_credit_card  car_park  gift_certificate  karaoke  \
1                           1         0                 0        0   
2                           0         0                 0        0   
4                           0         0                 0        0   
5                           1         1                 0        0   
8                           1         0                 1        0   
...                       ...       ...               ...      ...   
68333                       1         0                 0        1   
68334                       1         0                 1        1   
68335                       1         0                 1        1   
68336                       1         0                 1        1   
68338                       1         1                 0        0   

       free_delivery  music  pets  air_conditioning  view_on_ostankino_tower  \
1                  0      0     0                 0                        0   
2                  0      0     0                 0                        0   
4                  0      0     0                 0                        0   
5                  0      0     0                 0                        1   
8                  0      0     0                 0                        0   
...              ...    ...   ...               ...                      ...   
68333              0      0     0                 0                        0   
68334              0      0     0                 0                        0   
68335              1      0     0                 0                        0   
68336              1      0     0                 0                        0   
68338              0      0     0                 0                        0   

       average_bill_filna  
1                   500.0  
2                   500.0  
4                   500.0  
5                   500.0  
8                   500.0  
...                   ...  
68333                 0.0  
68334                 0.0  
68335                 0.0  
68336                 0.0  
68338                 0.0  

[43195 rows x 21 columns]

Заменили в таблице все значения большие 2500 на среднее значение

Задание номер 3

In [157]:
x = data[data["city"] == "msk"]
len(x)

28900

In [158]:
x = data[data["city"] == "spb"]
len(x)

14295

В Москве заведений чем в Санктпербурге

In [159]:
print(set(data[RUBRIC].tolist()))

{'Кальян-бар', 'Бар безалкогольных напитков', 'Булочная, пекарня', 'Столовая', 'Кофе с собой', 'Суши-бар', 'Спортбар', 'Бар, паб', 'Фудкорт', 'Кондитерская', 'Ресторан', 'Кофейня', 'Пиццерия', 'Кафе', 'Быстрое питание'}


In [160]:
print(len(set(data[RUBRIC].tolist())))

15


In [161]:
data[RUBRIC].value_counts()

Кафе                           10116
Ресторан                        8294
Быстрое питание                 5497
Кофейня                         4896
Бар, паб                        3051
Булочная, пекарня               2733
Кондитерская                    2516
Пиццерия                        1633
Кальян-бар                      1511
Столовая                        1460
Суши-бар                         756
Спортбар                         294
Кофе с собой                     289
Бар безалкогольных напитков      143
Фудкорт                            6
Name: rubric, dtype: int64

Всего 15 типо заведений
{'Кальян-бар', 'Бар безалкогольных напитков', 'Булочная, пекарня', 'Столовая', 'Кофе с собой', 'Суши-бар', 'Спортбар', 'Бар, паб', 'Фудкорт', 'Кондитерская', 'Ресторан', 'Кофейня', 'Пиццерия', 'Кафе', 'Быстрое питание'}

In [162]:
print(len(data[(data[RUBRIC] == 'Ресторан') | (data[RUBRIC] == 'Бар, паб')]))

11345


Всег ресторанов и пабов  всумме 11345

In [163]:
data_restoranse = data[(data[RUBRIC] == 'Ресторан') | (data[RUBRIC] == 'Бар, паб')]
print(sum(data_restoranse[AVERAGE_BILL] / len(data_restoranse[AVERAGE_BILL])))

825.0109594017605


средний чек ресторанов и пабов - 825.01


In [164]:
data.isna()

org_id   city  average_bill  rating  rubric  coffee_to_go  \
1       False  False         False   False   False         False   
2       False  False         False   False   False         False   
4       False  False         False   False   False         False   
5       False  False         False   False   False         False   
8       False  False         False   False   False         False   
...       ...    ...           ...     ...     ...           ...   
68333   False  False         False   False   False         False   
68334   False  False         False   False   False         False   
68335   False  False         False   False   False         False   
68336   False  False         False   False   False         False   
68338   False  False         False   False   False         False   

       food_delivery  breakfast  summer_terrace  wi_fi  ...  \
1              False      False           False  False  ...   
2              False      False           False  False  ...   
4              False      False           False  False  ...   
5              False      False           False  False  ...   
8              False      False           False  False  ...   
...              ...        ...             ...    ...  ...   
68333          False      False           False  False  ...   
68334          False      False           False  False  ...   
68335          False      False           False  False  ...   
68336          False      False           False  False  ...   
68338          False      False           False  False  ...   

       payment_by_credit_card  car_park  gift_certificate  karaoke  \
1                       False     False             False    False   
2                       False     False             False    False   
4                       False     False             False    False   
5                       False     False             False    False   
8                       False     False             False    False   
...                       ...       ...               ...      ...   
68333                   False     False             False    False   
68334                   False     False             False    False   
68335                   False     False             False    False   
68336                   False     False             False    False   
68338                   False     False             False    False   

       free_delivery  music   pets  air_conditioning  view_on_ostankino_tower  \
1              False  False  False             False                    False   
2              False  False  False             False                    False   
4              False  False  False             False                    False   
5              False  False  False             False                    False   
8              False  False  False             False                    False   
...              ...    ...    ...               ...                      ...   
68333          False  False  False             False                    False   
68334          False  False  False             False                    False   
68335          False  False  False             False                    False   
68336          False  False  False             False                    False   
68338          False  False  False             False                    False   

       average_bill_filna  
1                   False  
2                   False  
4                   False  
5                   False  
8                   False  
...                   ...  
68333               False  
68334               False  
68335               False  
68336               False  
68338               False  

[43195 rows x 21 columns]

In [165]:
print(len(data[(data[COFFEE_TO_GO] == 1) & (data[RUBRIC] == 'Кофе с собой')]) / len(data[data[RUBRIC] == 'Кофе с собой']))

0.3806228373702422


Приблизительно 0.38 весх заведений Кофе с собой действительно имеют фитчу кофе на вынос

In [166]:
data_msk = data[data["city"] == "msk"][AVERAGE_BILL]
data_spb = data[data["city"] == "spb"][AVERAGE_BILL]
average_msk = sum(data_msk) / len(data_msk)
average_spb = sum(data_spb) / len(data_spb)
print(abs(average_msk - average_spb))

29.437898517087774


Разницв между средним щначением средних чеков заведений в Москве и в Санктпетербурге
Разница маленькая по сравнению с средним значением всех чеков.

In [167]:
data.describe()

org_id  average_bill        rating  coffee_to_go  food_delivery  \
count  4.319500e+04  43195.000000  36438.000000  43195.000000   43195.000000   
mean   9.186257e+18    638.898085      4.112360      0.414608       0.501678   
std    5.326650e+18    306.075509      0.763436      0.492660       0.500003   
min    1.148366e+13    500.000000      0.000000      0.000000       0.000000   
25%    4.570691e+18    500.000000      3.827833      0.000000       0.000000   
50%    9.159396e+18    564.359301      4.260870      0.000000       1.000000   
75%    1.377324e+19    564.359301      4.600000      1.000000       1.000000   
max    1.844596e+19   2500.000000      5.000000      1.000000       1.000000   

          breakfast  summer_terrace         wi_fi  business_lunch  \
count  43195.000000    43195.000000  43195.000000    43195.000000   
mean       0.312629        0.359000      0.487279        0.370413   
std        0.463570        0.479713      0.499844        0.482921   
min        0.000000        0.000000      0.000000        0.000000   
25%        0.000000        0.000000      0.000000        0.000000   
50%        0.000000        0.000000      0.000000        0.000000   
75%        1.000000        1.000000      1.000000        1.000000   
max        1.000000        1.000000      1.000000        1.000000   

       payment_by_credit_card      car_park  gift_certificate       karaoke  \
count            43195.000000  43195.000000      43195.000000  43195.000000   
mean                 0.542771      0.029679          0.056002      0.166408   
std                  0.498173      0.169703          0.229928      0.372451   
min                  0.000000      0.000000          0.000000      0.000000   
25%                  0.000000      0.000000          0.000000      0.000000   
50%                  1.000000      0.000000          0.000000      0.000000   
75%                  1.000000      0.000000          0.000000      0.000000   
max                  1.000000      1.000000          1.000000      1.000000   

       free_delivery         music          pets  air_conditioning  \
count   43195.000000  43195.000000  43195.000000      43195.000000   
mean        0.036578      0.018891      0.002454          0.003218   
std         0.187726      0.136142      0.049477          0.056636   
min         0.000000      0.000000      0.000000          0.000000   
25%         0.000000      0.000000      0.000000          0.000000   
50%         0.000000      0.000000      0.000000          0.000000   
75%         0.000000      0.000000      0.000000          0.000000   
max         1.000000      1.000000      1.000000          1.000000   

       view_on_ostankino_tower  average_bill_filna  
count             43195.000000        4.319500e+04  
mean                  0.223174        5.643593e+02  
std                   0.416379        3.616570e+04  
min                   0.000000        0.000000e+00  
25%                   0.000000        0.000000e+00  
50%                   0.000000        0.000000e+00  
75%                   0.000000        5.000000e+02  
max                   1.000000        7.502000e+06

первое замечание, котрое можно сделать, это что средние чеки всез заведений лежат от 500 рублей до 2500 рублей
И что меньше половины заведений имеют фитчу кофе на вынос, изсходя из среднего значения по фитче coffee_to_go.
Также можно сделать вывод что пропуски значений были в фитчах rating и average_bill	(average_bill	мы уе заполнили пропуски средними значениями)


Задание 5
Создадим новую таблицу города СанктПетербург

In [171]:
data_spb = data[data["city"] == "spb"]

In [172]:
data_spb

org_id city  average_bill    rating           rubric  \
8      11913692756639492916  spb    500.000000  3.600000         Столовая   
9      17739408339297875651  spb    500.000000  4.276219  Быстрое питание   
10     12608002566984191599  spb    500.000000  3.428571          Кофейня   
11      1456099975270739091  spb    500.000000  4.748092         Бар, паб   
13      6647514245407759924  spb    500.000000  5.000000          Кофейня   
...                     ...  ...           ...       ...              ...   
68312   1308884143047589686  spb    564.359301       NaN             Кафе   
68317   2777831238439895201  spb    500.000000  5.000000         Ресторан   
68327   9047563910052896279  spb   1000.000000  3.927778         Ресторан   
68328  17662684569129497551  spb   1000.000000  4.561707         Ресторан   
68338    125079240973229511  spb    564.359301  4.633333         Ресторан   

       coffee_to_go  food_delivery  breakfast  summer_terrace  wi_fi  ...  \
8                 0              1          0               1      0  ...   
9                 1              1          1               1      1  ...   
10                1              1          1               0      1  ...   
11                0              1          1               0      0  ...   
13                1              1          1               1      1  ...   
...             ...            ...        ...             ...    ...  ...   
68312             0              0          0               0      1  ...   
68317             1              1          1               1      1  ...   
68327             1              1          1               1      1  ...   
68328             1              1          1               1      1  ...   
68338             0              0          0               0      1  ...   

       payment_by_credit_card  car_park  gift_certificate  karaoke  \
8                           1         0                 1        0   
9                           1         0                 0        0   
10                          0         0                 0        0   
11                          1         0                 0        0   
13                          0         0                 0        0   
...                       ...       ...               ...      ...   
68312                       1         1                 0        0   
68317                       1         0                 1        1   
68327                       1         0                 0        1   
68328                       1         0                 1        1   
68338                       1         1                 0        0   

       free_delivery  music  pets  air_conditioning  view_on_ostankino_tower  \
8                  0      0     0                 0                        0   
9                  0      0     0                 0                        0   
10                 0      0     0                 0                        0   
11                 0      0     0                 0                        0   
13                 0      0     0                 0                        0   
...              ...    ...   ...               ...                      ...   
68312              0      0     1                 1                        0   
68317              0      0     0                 0                        0   
68327              0      0     0                 0                        0   
68328              0      1     0                 0                        0   
68338              0      0     0                 0                        0   

       average_bill_filna  
8                   500.0  
9                   500.0  
10                  500.0  
11                  500.0  
13                  500.0  
...                   ...  
68312                 0.0  
68317               500.0  
68327              1000.0  
68328              1000.0  
68338                 0.0  

[14295 rows x 21 columns]

In [173]:
data_spb.describe()

org_id  average_bill        rating  coffee_to_go  food_delivery  \
count  1.429500e+04  14295.000000  12263.000000  14295.000000   14295.000000   
mean   9.135664e+18    619.202397      4.109644      0.434208       0.505841   
std    5.336384e+18    265.771969      0.758190      0.495670       0.499983   
min    9.955441e+14    500.000000      0.666667      0.000000       0.000000   
25%    4.515968e+18    500.000000      3.810345      0.000000       0.000000   
50%    9.117054e+18    564.359301      4.255172      0.000000       1.000000   
75%    1.373255e+19    564.359301      4.596154      1.000000       1.000000   
max    1.844596e+19   2500.000000      5.000000      1.000000       1.000000   

          breakfast  summer_terrace         wi_fi  business_lunch  \
count  14295.000000    14295.000000  14295.000000    14295.000000   
mean       0.331724        0.382162      0.505212        0.395383   
std        0.470849        0.485933      0.499990        0.488950   
min        0.000000        0.000000      0.000000        0.000000   
25%        0.000000        0.000000      0.000000        0.000000   
50%        0.000000        0.000000      1.000000        0.000000   
75%        1.000000        1.000000      1.000000        1.000000   
max        1.000000        1.000000      1.000000        1.000000   

       payment_by_credit_card      car_park  gift_certificate       karaoke  \
count            14295.000000  14295.000000      14295.000000  14295.000000   
mean                 0.549563      0.025044          0.060511      0.165233   
std                  0.497555      0.156263          0.238439      0.371403   
min                  0.000000      0.000000          0.000000      0.000000   
25%                  0.000000      0.000000          0.000000      0.000000   
50%                  1.000000      0.000000          0.000000      0.000000   
75%                  1.000000      0.000000          0.000000      0.000000   
max                  1.000000      1.000000          1.000000      1.000000   

       free_delivery         music          pets  air_conditioning  \
count   14295.000000  14295.000000  14295.000000      14295.000000   
mean        0.028611      0.016229      0.002448          0.002658   
std         0.166718      0.126361      0.049423          0.051492   
min         0.000000      0.000000      0.000000          0.000000   
25%         0.000000      0.000000      0.000000          0.000000   
50%         0.000000      0.000000      0.000000          0.000000   
75%         0.000000      0.000000      0.000000          0.000000   
max         1.000000      1.000000      1.000000          1.000000   

       view_on_ostankino_tower  average_bill_filna  
count                  14295.0        14295.000000  
mean                       0.0          345.715285  
std                        0.0          923.757719  
min                        0.0            0.000000  
25%                        0.0            0.000000  
50%                        0.0            0.000000  
75%                        0.0          500.000000  
max                        0.0        41500.000000

In [176]:
data.describe()

org_id  average_bill        rating  coffee_to_go  food_delivery  \
count  4.319500e+04  43195.000000  36438.000000  43195.000000   43195.000000   
mean   9.186257e+18    638.898085      4.112360      0.414608       0.501678   
std    5.326650e+18    306.075509      0.763436      0.492660       0.500003   
min    1.148366e+13    500.000000      0.000000      0.000000       0.000000   
25%    4.570691e+18    500.000000      3.827833      0.000000       0.000000   
50%    9.159396e+18    564.359301      4.260870      0.000000       1.000000   
75%    1.377324e+19    564.359301      4.600000      1.000000       1.000000   
max    1.844596e+19   2500.000000      5.000000      1.000000       1.000000   

          breakfast  summer_terrace         wi_fi  business_lunch  \
count  43195.000000    43195.000000  43195.000000    43195.000000   
mean       0.312629        0.359000      0.487279        0.370413   
std        0.463570        0.479713      0.499844        0.482921   
min        0.000000        0.000000      0.000000        0.000000   
25%        0.000000        0.000000      0.000000        0.000000   
50%        0.000000        0.000000      0.000000        0.000000   
75%        1.000000        1.000000      1.000000        1.000000   
max        1.000000        1.000000      1.000000        1.000000   

       payment_by_credit_card      car_park  gift_certificate       karaoke  \
count            43195.000000  43195.000000      43195.000000  43195.000000   
mean                 0.542771      0.029679          0.056002      0.166408   
std                  0.498173      0.169703          0.229928      0.372451   
min                  0.000000      0.000000          0.000000      0.000000   
25%                  0.000000      0.000000          0.000000      0.000000   
50%                  1.000000      0.000000          0.000000      0.000000   
75%                  1.000000      0.000000          0.000000      0.000000   
max                  1.000000      1.000000          1.000000      1.000000   

       free_delivery         music          pets  air_conditioning  \
count   43195.000000  43195.000000  43195.000000      43195.000000   
mean        0.036578      0.018891      0.002454          0.003218   
std         0.187726      0.136142      0.049477          0.056636   
min         0.000000      0.000000      0.000000          0.000000   
25%         0.000000      0.000000      0.000000          0.000000   
50%         0.000000      0.000000      0.000000          0.000000   
75%         0.000000      0.000000      0.000000          0.000000   
max         1.000000      1.000000      1.000000          1.000000   

       view_on_ostankino_tower  average_bill_filna  
count             43195.000000        4.319500e+04  
mean                  0.223174        5.643593e+02  
std                   0.416379        3.616570e+04  
min                   0.000000        0.000000e+00  
25%                   0.000000        0.000000e+00  
50%                   0.000000        0.000000e+00  
75%                   0.000000        5.000000e+02  
max                   1.000000        7.502000e+06

Среднее значение чека уменьшилось 638 до 619, уменьшение не значительное

количество заведений в  Санктпетербурге сильно меньше полвины всех заведений